File: euler_063.py
from time import time
from itertools import count

start = time()

solutions = set()

for n in count(1):
    added = 0
    for k in count(1):
        p = k**n
        if len(str(p)) < n:
            k += 1
        if len(str(p)) == n:
            solutions.add(p)
            added += 1
        if len(str(p)) > n:
            break
    if added == 0:
        break

print("Number of solutions: " + str(len(solutions)))


stop = time()
print("Time: " + str(stop - start))

File: euler_063.js
const start = Date.now();

let count = 1; // a = 1, 1^1 = 1

const numberOfDigits = (num) => {
    return Math.floor(Math.log10(num)) + 1;
}

// Analysis
//
// if a^n is an n-digit number, then 10^(n-1) <= a^n < 10^n
//
// taking log10: n-1 <= n*log(a) < n
//
// so a = 1, 2, 3, 4, 5, 6, 7, 8, 9 are the only options

for (const a of [2,3,4,5,6,7,8,9]) {
    let n = 1;
    while (numberOfDigits(a**n) < n + 1) {
        if (numberOfDigits(a**n) === n) {
            count++;
        }
        n++;
    }
}


const end = Date.now();

console.log("Execute time: ", end - start, " Number of integers is: ", count);